### ISSUE TITLE
AccessViolationException with MakeType(7, 3)

### ISSUE THREAD

**Issue by mpolicki** [shimat/opencvsharp#793]
# Summary of your issue
I was doing some experimenting with user-defined types. The example code throws an exception for `MakeType(7, 3)`. No exception for `CV_8UC4`.

> System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
>   Stack Trace: 
>     NativeMethods.core_MatExpr_toMat(IntPtr expr)
>     MatExpr.op_Implicit(MatExpr self) line 55
>     MatExpr.ToMat() line 72
> 

# Environment
WIndows 10 1903 18362.535 64-bit
VS 2019 16.4.1
.NET Framework 4.8.03752
OpenCvSharp4.Windows 4.1.1.20191110
## Example code:
```
            var matSize = new Size(4, 5);
            var matType = MatType.MakeType(7, 3);
            var mat = new Mat(matSize, matType);
            mat.NotEquals(mat).ToMat();
```

**Comment by shimat** [comment]
This is a problem of native OpenCV. `Mat.NotEquals` calls `cv::compare` internally. `cv::compare` does not currently support depth value `CV_16F` (7). 

cv::compare crashes at this line.
https://github.com/opencv/opencv/blob/master/modules/core/src/arithm.cpp#L1237

Because the result of `getCmpFunc()` is zero. `cmpTab[depth] == 0` when depth == 7
https://github.com/opencv/opencv/blob/master/modules/core/src/arithm.cpp#L1074

**Comment by shimat** [comment]
I made a pull request to opencv. https://github.com/opencv/opencv/pull/16208

**Comment by shimat** [comment]
This is fixed by OpenCV 4.2.0.
